### ISSUE TITLE
[FEATURE] google LYRA voice codec support

### ISSUE THREAD

**Issue by Slach** [mumble-voip/mumble#5154]
**Context**
Voice chat in very bad networks

**Describe the feature you have in mind**
To allow voice chat in very low throughput, could we add support for https://github.com/google/lyra?

**Comment by Kissaki** [comment]
How bad is “really bad”? My understanding is that Opus works well even in low bandwidth situations.

**Comment by ZeroAbility** [comment]
Since the technology appears to mainly be for Android, with Linux used for development and debugging, this may be a better request for the Android client. My own two cents, I am with @Kissaki that Opus should be sufficient. We already have two codecs we keep around for legacy support outside of Opus that are no longer supported. It would be best to reduce the codec dependency footprint before adding another.

**Comment by Slach** [comment]
ok. great. roger that

**Comment by poVoq** [comment]
Lyra is significantly lower bandwidth than opus though.

While I generally agree that adding another codec isn't the best idea, this could potentially be interesting for server admins hosting *a lot* of Mumble channels.

**Comment by JuniorJPDJ** [comment]
How about sound quality? Is there any comparison?

**Comment by poVoq** [comment]
The original announcement has some samples: https://ai.googleblog.com/2021/02/lyra-new-very-low-bitrate-codec-for.html

**Comment by JuniorJPDJ** [comment]
Oh, so it's machine learning based.
There's then a question about latency.
Quality is muuuch better.

**Comment by JuniorJPDJ** [comment]
>  this may be a better request for the Android client.

The protocol would first need to support it, then all clients would need to be able to at least decode it.

**Comment by ZeroAbility** [comment]
> The protocol would first need to support it, then all clients would need to be able to at least decode it.

The other issues are they don't support Linux for normal use, and they don't seem to have build instructions for Windows. It would make more sense to use something that would work on those platforms. Since it seems to mainly support Android, the use case might be peer-to-peer instead of client/server. I could definitely see advantages in low cellular coverage areas for a codec like that.

**Comment by davidebeatrici** [comment]
In that case it may make sense to implement support for it server-side in order to allow mobile clients such as Mumla to take advantage of it.

However, I would wait for a stable version first, in order to avoid incurring in another "CELT" situation.

**Comment by Krzmbrzl** [comment]
Absolutely. Before it's not stable we should definitely not include it.

Implementing it only server-side doesn't seem to make too much sense though as in that case it would only gain anything if all clients on the server were mobile ones that support LYRA

**Comment by 0ihsan** [comment]
lyra v2 released yesterday.
- https://opensource.googleblog.com/2022/09/lyra-v2-a-better-faster-and-more-versatile-speech-codec.html
- https://github.com/google/lyra
- https://news.ycombinator.com/item?id=33041114

**Comment by Krzmbrzl** [comment]
It seems that particularly they reduced the computational overhead so that the additional latency was significantly reduced. Now it seems at least somewhat viable to support it in Mumble (also support for Windows and macOS has been added)
